translate/translate.go

```diff
@@ -129,6 +134,85 @@ type Key struct {
 
 type informationElements map[Key]InformationElementEntry
 
+var reducedSizeErr error = errors.New("Unable to read reduced size encoding")
+
+// Helper method to read an unsigned reduced size field
+func reducedSizeReadUnsigned(bs []byte, maxBits int) (uint64, error) {
+	// Exit if `bs` has more bits than we can store
+	if len(bs)*8 > maxBits {
+		return 0, reducedSizeErr
+	}
+
+	switch len(bs) {
+	case 1:
+		return uint64(bs[0]), nil
+	case 2:
+		return uint64(binary.BigEndian.Uint16(bs)), nil
+	case 3:
+		return uint64(uint32(bs[0])<<16 + uint32(bs[1])<<8 + uint32(bs[2])), nil
+	case 4:
+		return uint64(binary.BigEndian.Uint32(bs)), nil
+	case 8:
+		return binary.BigEndian.Uint64(bs), nil
+	}
+	return 0, reducedSizeErr
+}
+
+// Helper method to read a signed reduced size field
+func reducedSizeReadSigned(bs []byte, maxBits int) (int64, error) {
```

Comment:
Is it possible to unit test this function?  Maybe just take the cases in the existing tests and put them through here.  I just feel like the signed one is most prone to errors, and testing would help